supervision/detection/utils.py

```diff
@@ -1481,3 +1481,102 @@ def box_iou_batch_with_jaccard(
         for d_idx, d in enumerate(boxes_detection):
             ious[d_idx, g_idx] = _jaccard(d, g, is_crowd[g_idx])
     return ious
+
+
+def remove_noisy_segments(
```

Comment:
please add unit tests to `remove_noisy_segments`; make sure test are parametrized; you can take inspiration from `test_mask_to_rle`